src/client/address_book/storage.cc

```diff
@@ -65,18 +67,34 @@ bool AddressBookStorage::GetAddress(
   return true;
 }
 
-void AddressBookStorage::AddAddress(
-    const kovri::core::IdentityEx& address) {
-  auto filename = GetAddressesPath() / (address.GetIdentHash().ToBase32() + ".b32");
-  std::ofstream file(filename.string(), std::ofstream::binary);
-  if (!file)
-    LOG(error) << "AddressBookStorage: can't open file " << filename;
+void AddressBookStorage::AddAddress(const kovri::core::IdentityEx& address)
+{
+  auto filename =
+      GetAddressesPath() / (address.GetIdentHash().ToBase32() + ".b32");
+  kovri::core::OutputFileStream file(filename.string(), std::ofstream::binary);
+  if (!file.Good())
+    {
+      m_Exception.Dispatch(__func__);
```

Comment:
This is useless because nothing is thrown before it. Exception dispatchers are meant to be used when catching. If you do more than one check, catching would be appropriate. Otherwise, leave it to the caller.